internal/core/metadata/container/config.go

```diff
@@ -27,3 +28,17 @@ var ConfigurationName = di.TypeInstanceToName((*config.ConfigurationStruct)(nil)
 func ConfigurationFrom(get di.Get) *config.ConfigurationStruct {
 	return get(ConfigurationName).(*config.ConfigurationStruct)
 }
+
+// SecretProviderName contains the name of the interfaces.SecretProvider implementation in the DIC.
+var SecretProviderName = di.TypeInstanceToName((*interfaces.SecretProvider)(nil))
+
+// SecretProviderFrom helper function queries the DIC and returns the interfaces.SecretProvider
+// implementation.
+func SecretProviderFrom(get di.Get) interfaces.SecretProvider {
+	provider, ok := get(SecretProviderName).(interfaces.SecretProvider)
+	if !ok {
+		return nil
+	}
+
+	return provider
+}
```

Comment:
This already exist here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/bootstrap/container/secret.go#L28